hadoop-ozone/s3gateway/src/test/java/org/apache/hadoop/ozone/s3/endpoint/TestObjectHead.java

```diff
@@ -185,4 +192,97 @@ public void testHeadWhenKeyIsAFileAndKeyPathEndsWithASlash()
     assertEquals(HttpStatus.SC_NOT_FOUND, response.getStatus());
     bucket.deleteKey(keyPath);
   }
+
+  @Test
+  public void testHeadWithRangeHeader() throws Exception {
+    //GIVEN
+    String value = "0123456789";
+    OzoneOutputStream out = bucket.createKey("key1",
+        value.getBytes(UTF_8).length,
+        ReplicationConfig.fromTypeAndFactor(ReplicationType.RATIS,
+        ReplicationFactor.ONE), new HashMap<>());
+    out.write(value.getBytes(UTF_8));
+    out.close();
```

Comment:
Please move key creation to `setup()` to avoid duplication.  While at that, please change to:

```java
try (OutputStream out = ...) {
  out.write(...)
}
```